app/Repositories/Cts/MentorRepository.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Repositories\Cts;
+
+use App\Models\Cts\PositionValidation;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Collection;
+
+class MentorRepository
+{
+    public function getMentorsWithin(int $rtsId): Collection
+    {
+        $positionValidations = PositionValidation::with(['member', 'position'])
+        ->whereHas('position', function (Builder $query) use ($rtsId) {
+            $query->where('rts_id', '=', $rtsId);
+        })->where('status', '=', 5)->get();
+
+        $mentors = collect();
+
+        foreach ($positionValidations as $positions) {
+            $mentors->push($positions->member);
+        }
+
+        return $this->format($mentors->unique());
+    }
+
+    public function getMentorsFor(string $string): Collection
```

Comment:
How long is a piece of the variable string? (What is `$string`?)

Reply:
Would search be a better term? A user can pass either a position (EGKK_GND) or a callsign (EGKK) as we just do a LIKE query in the method.

Reply:
Aye it would be

Reply:
Done! f6b199a5